### PR TITLE
K8SPS-283 update haproxy in docs

### PR DESCRIPTION
**Pull request by fiowro**

### CODE REVIEW

File: docs/haproxy-conf.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+* `cluster1-haproxy` service listening on ports 3306 (MySQL) and 3309 (the [proxy protocol](https://www.haproxy.com/blog/haproxy/proxy-protocol/)).
+    This service is pointing to the MySQL cluster member number zero 
+    (`cluster1-mysql-0`) by default when this member is available. If a zero
+    member is not available, members are selected in descending order of their
+    numbers (e.g. `cluster1-mysql-2`, then `cluster1-mysql-1`, etc.). This
+    service can be used for both read and write load, or it can also be used
+    just for write load (single writer mode) in setups with split write and read
+    loads.
+
+* `cluster1-mysql-proxy` service selects MySQL cluster members to serve queries
```
Comment (fiowro) [inline docs/haproxy-conf.md:32]: Needs factchecking

File: docs/haproxy-conf.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
+# Configuring Load Balancing with HAProxy
+
+Percona Operator for MySQL provides load balancing and proxy service with
+[HAProxy](https://haproxy.org) (enabled by default). HAProxy is the only
+solution proxy when asynchronous replication betweem MySQL instances is turned,
```
Comment (tplavcic) [inline docs/haproxy-conf.md:5]:
```suggestion
solution proxy when asynchronous replication between MySQL instances is enabled,
```

File: docs/haproxy-conf.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    When enabling HAProxy, make sure MySQL Router is disabled
+    (`proxy.router.enabled` option should be set to `false`).
+
+For example, you can use the following command to enable HAProxy for existing
+cluster:
+
+```bash
+$ kubectl patch ps cluster1 --type=merge --patch '{
+  "spec": {
+     "router": {
```
Comment (tplavcic) [inline docs/haproxy-conf.md:21]:
```suggestion
     "proxy": {
```

File: docs/haproxy-conf.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+a zero member is not available, members are selected in descending order of
+their numbers (e.g. `cluster1-mysql-2`, then `cluster1-mysql-1`, etc.). It can
+be used for both read and write load, or it can also be used just for write load
+(single writer mode) in setups with split write and read loads. On 3307 port
+this service selects MySQL cluster members to serve queries following the Round
+Robin load balancing algorithm. 
+
+When the cluster with HAProxy is upgraded, the following steps
+take place. First, reader members are upgraded one by one: the Operator waits
+until the upgraded Percona Distribution for MySQL luster member becomes synced,
```
Comment (tplavcic) [inline docs/haproxy-conf.md:46]:
```suggestion
until the upgraded Percona Distribution for MySQL cluster member becomes synced,
```

File: docs/haproxy-conf.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+You can pass custom configuration to HAProxy, adding options from the
+[haproxy.cfg](https://www.haproxy.com/blog/the-four-essential-sections-of-an-haproxy-configuration/)
+configuration file to the  `haproxy.configuration` Custom Resource option in
+the `deploy/cr.yaml` file. Here is an example:
+
+```yaml
+...
+haproxy:
+    enabled: true
```
Comment (tplavcic) [inline docs/haproxy-conf.md:60]: I think usually we use two spaces indentation, but I might be wrong, please check.

**Review by tplavcic — CHANGES_REQUESTED** [review]

**Pull request merged**